### PR TITLE
Feature/PIV-24/asset-uploader

### PR DESCRIPTION
**Pull request by RamsesDijkstra**
Contains asset upload EP, and a small refactor of existing terminology

### CODE REVIEW

File: cmd/service/main.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ func run() error {
 func createNode(ctx context.Context, cfg *config.Config, db database.DB) (universe.Node, error) {
 	worlds := worlds.NewWorlds(db)
 	assets2d := assets_2d.NewAssets2d(db)
-	assets3d := assets_3d.NewAssets3d(db)
+	assets3d := assets_3d.NewAssets3d(cfg, db)
```
Comment (cnaize) [inline cmd/service/main.go]: please do it in different way
we have the config in our main context
look https://github.com/momentum-xyz/ubercontroller/blob/develop/universe/worlds/worlds.go#L44-L47

File: universe/assets_3d/api_assets_3d.go [hunk 2]
```diff
@@ -38,11 +40,11 @@ func (a *Assets3d) apiGetAssets3d(c *gin.Context) {
 	var a3dMap map[uuid.UUID]universe.Asset3d
 	predicateFn := func(asset3dID uuid.UUID, asset3d universe.Asset3d) bool {
 		meta := asset3d.GetMeta()
-		kind := utils.GetFromAnyMap(*meta, "kind", "")
-		return kind == inQuery.Kind
+		category := utils.GetFromAnyMap(*meta, "type", "")
```
Comment (cnaize) [inline universe/assets_3d/api_assets_3d.go]: what if meta == nil?

File: universe/assets_3d/assets_3d.go [hunk 3]
```diff
@@ -20,13 +21,15 @@ var _ universe.Assets3d = (*Assets3d)(nil)
...
 type Assets3d struct {
 	ctx    context.Context
+	cfg    *config.Config
 	log    *zap.SugaredLogger
 	db     database.DB
 	assets *generic.SyncMap[uuid.UUID, universe.Asset3d]
 }
 
-func NewAssets3d(db database.DB) *Assets3d {
+func NewAssets3d(cfg *config.Config, db database.DB) *Assets3d {
```
Comment (cnaize) [inline universe/assets_3d/assets_3d.go]: fix please

File: universe/common/api/dto/types.go [hunk 4]
```diff
@@ -1,5 +1,12 @@
 package dto
 
+type Asset3dType int8
+
+const (
+	AddressableAssetType Asset3dType = 0
+	GLTFAsset3dType      Asset3dType = 1
```
Comment (cnaize) [inline universe/common/api/dto/types.go]: I think it's better to use iota

File: universe/assets_3d/api_assets_3d.go [hunk 5]
```diff
@@ -113,6 +117,99 @@ func (a *Assets3d) apiAddAssets3d(c *gin.Context) {
...
+		err := errors.WithMessage(err, "Assets3d: apiUploadAsset3d: failed to post data to media-manager")
+		api.AbortRequest(c, http.StatusBadRequest, "failed_to_post_request", err, a.log)
+		return
+	}
+
+	defer resp.Body.Close()
+
+	response := dto.HashResponse{}
+
+	errs := json.NewDecoder(resp.Body).Decode(&response)
```
Comment (boyan-d) [inline universe/assets_3d/api_assets_3d.go]: cosmetic change:  you can inline this `if errs := ... ; errs != nil {`

File: universe/assets_3d/api_assets_3d.go [hunk 6]
```diff
@@ -37,12 +41,16 @@ func (a *Assets3d) apiGetAssets3d(c *gin.Context) {
 
 	var a3dMap map[uuid.UUID]universe.Asset3d
 	predicateFn := func(asset3dID uuid.UUID, asset3d universe.Asset3d) bool {
+		var category string
 		meta := asset3d.GetMeta()
-		kind := utils.GetFromAnyMap(*meta, "kind", "")
-		return kind == inQuery.Kind
+
+		if meta != nil {
+			category = utils.GetFromAnyMap(*meta, "type", "")
```
Comment (cnaize) [inline universe/assets_3d/api_assets_3d.go]: maybe just return false if meta == nil?

File: universe/assets_3d/api_assets_3d.go [hunk 7]
```diff
@@ -113,6 +123,105 @@ func (a *Assets3d) apiAddAssets3d(c *gin.Context) {
...
+	}
+
+	defer openedFile.Close()
+
+	cfg := utils.GetFromAny(a.ctx.Value(types.ConfigContextKey), (*config.Config)(nil))
+	if cfg == nil {
+		err = errors.WithMessage(err, "Assets3d: apiUploadAsset3d: failed to get config from context")
+		api.AbortRequest(c, http.StatusInternalServerError, "failed_to_get_asset3d", err, a.log)
+		return
+	}
```
Comment (cnaize) [inline universe/assets_3d/api_assets_3d.go]: I think it's better to move to Initialize() to get rid of get cfg repeats

**Review by cnaize — APPROVED** [review]

**Review by boyan-d — APPROVED** [review]

**Pull request merged**